pkg/controllers/metrics/pod/controller.go

```diff
@@ -326,6 +326,22 @@ func (c *Controller) recordPodStartupMetric(pod *corev1.Pod, schedulableTime tim
 			c.pendingPods.Delete(key)
 			// Clear cluster state's representation of these pods as we don't need to keep track of them anymore
 			c.cluster.ClearPodSchedulingMappings(client.ObjectKeyFromObject(pod))
+		} else if podutils.IsTerminal(pod) {
+			// We do not emit the startup duration metric for pods that are terminal because such pods will have
+			// Ready status condition set to False which will cause the metric to take negative values.
```

Comment:
I'd specifically call out that the last transition time was updated when it transitioned to false after the pod went terminal

Reply:
But in this case more than that lastTransitionTime, it's the fact that condition stores Ready condition if it's set to true. If not then ideally there's no valid lastTransitionTime stored here -
```
cond, ready := lo.Find(pod.Status.Conditions, func(c corev1.PodCondition) bool {
return c.Type == corev1.PodReady && c.Status == corev1.ConditionTrue
})
```